api/app/Http/Controllers/ChildEntityController.php

```diff
@@ -159,17 +189,14 @@ public function putMany($id, Request $request)
             if ($id && !empty($childModels) && $childModels->has($id)) {
                 $childModel = $childModels->get($id);
             } else {
-                $childModel = $relation->getQuery()->getModel()->newInstance();
-                ;
+                $childModel = $this->getChildModel()->newInstance();
             }
-            /** @var BaseModel $childModel */
+            /** @var ChildBaseModel $childModel */
             $childModel->fill($requestEntity);
+            $childModel->attachParent($model);
             $putModels[] = $childModel;
         }
-        $putModels = $relation->getQuery()->getModel()->newCollection($putModels);
-
-        $model->setRelation($this->relationName, $putModels, $relation);
-        $this->saveModel($model);
+        ModelHelper::saveMany($putModels);
```

Comment:
In this case we should probably use the relationship to saveMany as we may have defined more constraints. I think the child controller needs to be relationship aware.

So this command would be something more like `$model->{self::relationship}->save($putModels);`